ecmd-core/ext/cip/capi/cipClientCapi.H

```diff
@@ -1059,6 +1059,82 @@ uint32_t cipPutMemProcVariousAddrType(ecmdChipTarget & i_target, ecmdDataBuffer
 //@}
 /* End Memory Functions */
 
+/** @name Breakpoint Functions */
+//@{
+#ifndef CIP_REMOVE_BREAKPOINT_FUNCTIONS
+/**
+ @brief Set a software breakpoint in Processor 
+ @param i_target Struct that contains chip and cage/node/slot/position/core/thread information
+ @param i_address Address to set breakpoint at
+ @param i_xlateVars Scruct with numerous translation variables
+ @param o_brkptTableEntries List of cipBrkptTableEntry structs
+ @param o_virtualAddress Caculated Virtual Address
+ @retval ECMD_TARGET_NOT_CONFIGURED if target is not available in the system
+ @retval ECMD_RING_CACHE_ENABLED Ring Cache enabled function - must be disabled to use this function
+ @retval ECMD_CLOCKS_IN_INVALID_STATE Chip Clocks were in an invalid state to perform the operation
+ @retval ECMD_RETRY_WITH_VIRTUAL_ADDR Requests that the user retries operation with returned Virtual Address
+ @retval ECMD_SUCCESS if successful
+ @retval nonzero if unsuccessful
+
+ TARGET DEPTH  : Thread<br>
+ TARGET STATES : Unused<br>
+
+*/
+uint32_t cipSetSoftwareBreakpoint (const ecmdChipTarget & i_target, const ecmdDataBuffer & i_address, cipXlateVariables i_xlateVars, std::list<cipBrkptTableEntry> & o_brkptTableEntries, ecmdDataBuffer & o_virtualAddress);
```

Comment:
Should i_xlateVars be const for all these new APIs?

Reply:
sure, it could.

Reply:
done